float64[] reference
float64 goal_time
